packages/cli/src/PublicApi/v1/Credentials/credentials.service.ts

```diff
@@ -0,0 +1,126 @@
+import { FindOneOptions } from 'typeorm';
+import { UserSettings, Credentials } from 'n8n-core';
+import { Db, ICredentialsDb } from '../../..';
+import { CredentialsEntity } from '../../../databases/entities/CredentialsEntity';
+import { SharedCredentials } from '../../../databases/entities/SharedCredentials';
+import { validateEntity } from '../../../GenericHelpers';
+import { User } from '../../../databases/entities/User';
+
+export async function getCredentials(
+	credentialId: number | string,
+): Promise<ICredentialsDb | undefined> {
+	return Db.collections.Credentials.findOne(credentialId);
+}
+
+export async function getSharedCredentials(
+	userId: string,
+	credentialId: number | string,
+	relations?: string[],
+): Promise<SharedCredentials | undefined> {
+	const options: FindOneOptions = {
+		where: {
+			user: { id: userId },
+			credentials: { id: credentialId },
+		},
+	};
+
+	if (relations) {
+		options.relations = relations;
+	}
+
+	return Db.collections.SharedCredentials.findOne(options);
+}
+
+export async function createCredential(
+	properties: Partial<CredentialsEntity>,
+): Promise<CredentialsEntity> {
+	const newCredential = new CredentialsEntity();
+
+	Object.assign(newCredential, properties);
+
+	if (!newCredential.nodesAccess || newCredential.nodesAccess.length === 0) {
+		newCredential.nodesAccess = [
+			{
+				nodeType: `n8n-nodes-base.${properties.type?.toLowerCase() ?? 'unknown'}`,
+				date: new Date(),
+			},
+		];
+	} else {
+		// Add the added date for node access permissions
+		newCredential.nodesAccess.forEach((nodeAccess) => {
+			// eslint-disable-next-line no-param-reassign
+			nodeAccess.date = new Date();
+		});
+	}
+
+	await validateEntity(newCredential);
```

Comment:
Should not we leave this validation to the openapi spec? The library we are using takes care of the validation of whatever we define in the spec and provides a helpful error message in case something is wrong.